src/main/java/org/snomed/aag/data/services/ProjectAcceptanceCriteriaService.java

```diff
@@ -133,34 +136,29 @@ public ProjectAcceptanceCriteria findByBranchPathAndProjectIterationOrThrow(Stri
     }
 
     /**
-     * Find project acceptance criteria for the latest iteration from this or the parent branch.
-	 * Add all current mandatory criteria items.
+     * Return ProjectAcceptanceCriteria with relevant CriteriaItems. If no ProjectAcceptanceCriteria can be found from the given
+     * branchPath, the parent Branch will be queried.
      *
-     * @param branchPath Field to match in query.
-     * @return The project acceptance criteria or null if none found.
+     * @param branchPath Branch path to query for ProjectAcceptanceCriteria
+     * @return ProjectAcceptanceCriteria with relevant CriteriaItems
      */
-    public ProjectAcceptanceCriteria findEffectiveCriteriaWithMandatoryItems(String branchPath) {
-		ProjectAcceptanceCriteria criteria = getLatestProjectAcceptanceCriteria(branchPath);
-
+    public ProjectAcceptanceCriteria findByBranchPathWithRelevantCriteriaItems(String branchPath) {
+        ProjectAcceptanceCriteria criteria = getFromBranchOrParent(branchPath);
         if (criteria == null) {
-            String parentPath = PathUtil.getParentPath(branchPath);
-            if (parentPath != null) {
-                criteria = getLatestProjectAcceptanceCriteria(parentPath);
-            }
-
-            if (criteria == null) {
-                return null;
-            }
+            return null;
```

Comment:
This method has changed a lot, and the diff makes it difficult to follow. 

Now:- essentially, all relevant CriteriaItems for a ProjectAcceptanceCriteria are read. Then, a CriteriaItem is removed from the ProjectAcceptanceCriteria if it results in a flag conflict against the Branch metadata.